pkg/util/lazyregexp/re.go

```diff
@@ -0,0 +1,93 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package lazyregexp holds lazy initliazed regexp code
+package lazyregexp
+
+import (
+	"regexp"
+	"sync"
+)
+
+var inTest bool
+
+// LazyRegexp is a lazy initialized regexp
+type LazyRegexp struct {
+	expression string
+	once       sync.Once
+	compiled   *regexp.Regexp
+}
+
+// New returns a new LazyRegexp
+func New(expression string) *LazyRegexp {
+	lr := &LazyRegexp{
+		expression: expression,
+	}
+
+	// in test code, we force the regexp to be compiled at creation time
+	// to make sure the pattern is valid
```

Comment:
Do we actually want this? I would expect any tests to also validate the lazy-loading functionality, so if test code behaves differently than production code then the tests are invalid.